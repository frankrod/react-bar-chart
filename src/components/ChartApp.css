/* .sliderBox {
  width: 300px;
 
} */

.container { 
  display: flex;
  flex-direction: column;
  align-items: center;
}

.item {
  width: 500px;
  margin-top: 100px;
}