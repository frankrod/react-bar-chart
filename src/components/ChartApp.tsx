import * as React from 'react';
import { Bar } from 'react-chartjs-2';
import InputRange, { Range } from 'react-input-range';

import 'react-input-range/lib/css/index.css';
import './ChartApp.css'; 

interface Props {
  data: number[];
  rangeSliderValue: Range; 
  onChangeRangeSliderValue: (rangeSliderValue: number | Range) => void;
}

const range = (start: number, end: number) => Array(end - start + 1).fill(undefined).map((_, index) => index + start)

const ChartApp = ({ data, rangeSliderValue, onChangeRangeSliderValue }: Props) => {
  
  const min = 1;
  const max = data.length;
  const includedBarColor = 'RGB(49, 172, 170, 0.8)';
  const excludedBarColor = 'RGB(153, 153, 153, 0.8)';

  const barsFiltered = range(rangeSliderValue.min, rangeSliderValue.max);

  const barData = {
    labels: data.map((_, index) => `${index+1}`),
    datasets: [
      {
        label: 'Bar chart',
        backgroundColor: data.map((_, index) => barsFiltered.includes(index+1) ? includedBarColor : excludedBarColor),
        borderWidth: 1,
        data: data.sort((a, b) => b-a)
      },
    ]
  };

  return (
    <div className="container" >
      <div className="item" >
        <Bar
          data={barData}
          height={400}
          options={{
            maintainAspectRatio: false
          }}
        />
      </div>
      <div className="item" >
        <InputRange
          maxValue={max}
          minValue={min}
          value={rangeSliderValue}
          onChange={value => onChangeRangeSliderValue(value)} 
        />
      </div>
    </div> 
  );
}

export default ChartApp;
