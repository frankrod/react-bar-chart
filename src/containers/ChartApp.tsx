import ChartApp from '../components/ChartApp';
import * as actions from '../actions/chartApp';
import { State } from '../reducers';
import { Range } from 'react-input-range';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';

// export const mapStateToProps = ({ rangeSliderValue }: ChartAppState) => ({
//   rangeSliderValue
// });

export const mapStateToProps = ({ chartApp }: State) => {
  return  {
    rangeSliderValue: chartApp.rangeSliderValue
  }
};

export const mapDispatchToProps = (dispatch: Dispatch<actions.Action>) => {
  return {
    onChangeRangeSliderValue: (value: Range) => dispatch(actions.onChangeRangeSliderValue(value))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ChartApp);