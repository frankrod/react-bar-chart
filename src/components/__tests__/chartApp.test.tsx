import * as React from 'react';
import { shallow } from 'enzyme';
import ChartApp from '../ChartApp';

import '../ChartApp.css'; 

describe('<ChartApp />', () => {
  const props = {
    rangeSliderValue: { min:1, max: 8 },
    onChangeRangeSliderValue: jest.fn()
  };

  it('should render successfully', () => {
    const wrapper = shallow(<ChartApp { ...props } />);

    expect(wrapper).toMatchSnapshot();
  });
});