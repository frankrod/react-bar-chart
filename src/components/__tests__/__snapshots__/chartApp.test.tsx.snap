// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartApp /> should render successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChartApp
    data={
        Array [
            3,
            3,
            1,
          ]
    }
    onChangeRangeSliderValue={[Function]}
    rangeSliderValue={
        Object {
            "max": 8,
            "min": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="item"
>
          <Bar
                    data={
                              Object {
                                        "datasets": Array [
                                          Object {
                                            "backgroundColor": Array [
                                              "RGB(49, 172, 170, 0.8)",
                                              "RGB(49, 172, 170, 0.8)",
                                              "RGB(49, 172, 170, 0.8)",
                                            ],
                                            "borderWidth": 1,
                                            "data": Array [
                                              3,
                                              3,
                                              1,
                                            ],
                                            "label": "Bar chart",
                                          },
                                        ],
                                        "labels": Array [
                                          "1",
                                          "2",
                                          "3",
                                        ],
                                      }
                    }
                    height={400}
                    options={
                              Object {
                                        "maintainAspectRatio": false,
                                      }
                    }
          />
</div>,
        <div
          className="item"
>
          <Unknown
                    maxValue={3}
                    minValue={1}
                    onChange={[Function]}
                    value={
                              Object {
                                        "max": 8,
                                        "min": 1,
                                      }
                    }
          />
</div>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Bar
            data={
                        Object {
                                    "datasets": Array [
                                      Object {
                                        "backgroundColor": Array [
                                          "RGB(49, 172, 170, 0.8)",
                                          "RGB(49, 172, 170, 0.8)",
                                          "RGB(49, 172, 170, 0.8)",
                                        ],
                                        "borderWidth": 1,
                                        "data": Array [
                                          3,
                                          3,
                                          1,
                                        ],
                                        "label": "Bar chart",
                                      },
                                    ],
                                    "labels": Array [
                                      "1",
                                      "2",
                                      "3",
                                    ],
                                  }
            }
            height={400}
            options={
                        Object {
                                    "maintainAspectRatio": false,
                                  }
            }
/>,
          "className": "item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Object {
              "datasets": Array [
                Object {
                  "backgroundColor": Array [
                    "RGB(49, 172, 170, 0.8)",
                    "RGB(49, 172, 170, 0.8)",
                    "RGB(49, 172, 170, 0.8)",
                  ],
                  "borderWidth": 1,
                  "data": Array [
                    3,
                    3,
                    1,
                  ],
                  "label": "Bar chart",
                },
              ],
              "labels": Array [
                "1",
                "2",
                "3",
              ],
            },
            "height": 400,
            "options": Object {
              "maintainAspectRatio": false,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Unknown
            maxValue={3}
            minValue={1}
            onChange={[Function]}
            value={
                        Object {
                                    "max": 8,
                                    "min": 1,
                                  }
            }
/>,
          "className": "item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "maxValue": 3,
            "minValue": 1,
            "onChange": [Function],
            "value": Object {
              "max": 8,
              "min": 1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": undefined,
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="item"
>
            <Bar
                        data={
                                    Object {
                                                "datasets": Array [
                                                  Object {
                                                    "backgroundColor": Array [
                                                      "RGB(49, 172, 170, 0.8)",
                                                      "RGB(49, 172, 170, 0.8)",
                                                      "RGB(49, 172, 170, 0.8)",
                                                    ],
                                                    "borderWidth": 1,
                                                    "data": Array [
                                                      3,
                                                      3,
                                                      1,
                                                    ],
                                                    "label": "Bar chart",
                                                  },
                                                ],
                                                "labels": Array [
                                                  "1",
                                                  "2",
                                                  "3",
                                                ],
                                              }
                        }
                        height={400}
                        options={
                                    Object {
                                                "maintainAspectRatio": false,
                                              }
                        }
            />
</div>,
          <div
            className="item"
>
            <Unknown
                        maxValue={3}
                        minValue={1}
                        onChange={[Function]}
                        value={
                                    Object {
                                                "max": 8,
                                                "min": 1,
                                              }
                        }
            />
</div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Bar
              data={
                            Object {
                                          "datasets": Array [
                                            Object {
                                              "backgroundColor": Array [
                                                "RGB(49, 172, 170, 0.8)",
                                                "RGB(49, 172, 170, 0.8)",
                                                "RGB(49, 172, 170, 0.8)",
                                              ],
                                              "borderWidth": 1,
                                              "data": Array [
                                                3,
                                                3,
                                                1,
                                              ],
                                              "label": "Bar chart",
                                            },
                                          ],
                                          "labels": Array [
                                            "1",
                                            "2",
                                            "3",
                                          ],
                                        }
              }
              height={400}
              options={
                            Object {
                                          "maintainAspectRatio": false,
                                        }
              }
/>,
            "className": "item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "data": Object {
                "datasets": Array [
                  Object {
                    "backgroundColor": Array [
                      "RGB(49, 172, 170, 0.8)",
                      "RGB(49, 172, 170, 0.8)",
                      "RGB(49, 172, 170, 0.8)",
                    ],
                    "borderWidth": 1,
                    "data": Array [
                      3,
                      3,
                      1,
                    ],
                    "label": "Bar chart",
                  },
                ],
                "labels": Array [
                  "1",
                  "2",
                  "3",
                ],
              },
              "height": 400,
              "options": Object {
                "maintainAspectRatio": false,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Unknown
              maxValue={3}
              minValue={1}
              onChange={[Function]}
              value={
                            Object {
                                          "max": 8,
                                          "min": 1,
                                        }
              }
/>,
            "className": "item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "maxValue": 3,
              "minValue": 1,
              "onChange": [Function],
              "value": Object {
                "max": 8,
                "min": 1,
              },
            },
            "ref": null,
            "rendered": null,
            "type": undefined,
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
